fastmlx/fastmlx.py

```diff
@@ -88,6 +88,16 @@ class ChatCompletionResponse(BaseModel):
 app = FastAPI()
 
 
+# Custom type function
+def int_or_float(value):
```

Comment:
Can we rename this to `parse_int_or_float` or `parse_numeric` ?

and refactor it to not use loops.
```suggestion
def int_or_float(value):
    """
    Convert string to int or float, or raise ArgumentTypeError.
    """
    try:
        return int(value)
    except ValueError:
        try:
            return float(value)
        except ValueError:
            raise argparse.ArgumentTypeError(f"{value} is not a valid number")
```

Reply:
It's being used as a type indicator while adding args (https://github.com/SiddhantSadangi/fastmlx/blob/8c3a7b5911f872c87def720888fca549019cd032/fastmlx/fastmlx.py#L297), so I think this name makes more sense in that context (`type=int_or_float` instead of `type=parse_*`).

Regarding refactoring, done ✅

Reply:
Makes sense!